torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -595,11 +632,17 @@ TensorView* TensorView::cache_after() {
     setComputeAt(consumer, this_ca_pos, this_ca_pos);
     consumer->setComputeAt(this_ca_view, this_ca_pos, rel_ca_pos);
   } else {
+    bool cache_replayed = false;
     // Check users of this TV for computeAt for cache_after on inputs
     for (auto expr : fusion()->unordered_uses(consumer)) {
       for (TensorView* output :
            ir_utils::filterByType<TensorView>(expr->outputs())) {
         if (output->hasComputeAt()) {
+          if (!cache_replayed) {
+            // Completely transform consumer according to output
+            TransformReplay::replayPasC(consumer, output, -1);
```

Comment:
We have transformations to merge non-reduction and reduction axes together. Those transformations need to be replayed for all dimensions, otherwise I need to manually replay them.